unblob/finder.py

```diff
@@ -2,166 +2,202 @@
 Searching Chunk related functions.
 The main "entry point" is search_chunks_by_priority.
 """
-import io
+from enum import Flag
 from functools import lru_cache
-from operator import itemgetter
-from pathlib import Path
-from typing import Dict, List, Tuple, Type
+from typing import Dict, List, Optional, Tuple, Type
 
-import yara
+import attr
+import hyperscan
 from structlog import get_logger
 
-from .file_utils import InvalidInputFormat, LimitedStartReader
+from .file_utils import InvalidInputFormat
 from .handlers import Handlers
 from .logging import noformat
-from .models import Handler, TaskResult, ValidChunk, YaraMatchResult
+from .models import File, Handler, TaskResult, ValidChunk
 from .report import CalculateChunkExceptionReport
 
 logger = get_logger()
 
 
-_YARA_RULE_TEMPLATE = """
-rule {NAME}
-{{
-    {YARA_RULE}
-}}
-"""
+@attr.define
+class HyperscanMatchContext:
+    handler_map: Dict[int, Handler]
+    file: File
+    file_size: int
+    all_chunks: List
+    task_result: TaskResult
+
+
+class ScanContinue(Flag):
+    Continue = False
+    Terminate = True
```

Comment:
I feel the enum should be named `ScanTerminate` instead, as it is a surprising definition: the value of `ScanContinue.Continue` is `False`.

Reply:
Maybe another option would be to prefix with `_HyperScan` to make it
- module private
- explain, that it is used by Hyperscan only.